sklearn/tree/_export.py

```diff
@@ -79,7 +79,7 @@ def __repr__(self):
         "decision_tree": [DecisionTreeClassifier, DecisionTreeRegressor],
         "max_depth": [Interval(Integral, 0, None, closed="left"), None],
         "feature_names": [list, None],
-        "class_names": [list, None],
+        "class_names": [list, None, "boolean"],
```

Comment:
While we're here, we should probably also change `list` to `"array-like"` similarly to `export_graphviz`. Also, totally just a nit-pick, but I like keeping `None` as the final option by convention :)

```suggestion
        "class_names": ["array-like", "boolean", None],
```